modules/java_api/src/main/java/org/intel/openvino/PartialShape.java

```diff
@@ -0,0 +1,39 @@
+// Copyright (C) 2020-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.intel.openvino;
+
+/** This class represents the definitions and operations about partial shape. */
+public class PartialShape extends Wrapper {
+
+    public PartialShape(long addr) {
+        super(addr);
+    }
+
+    /**
+     * Get the dimension at specified index of a partial shape.
+     *
+     * @param index The index of dimension.
+     * @return The particular dimension of partial shape.
+     */
+    public long get_dimension(int index) {
```

Comment:
Please align with C++ API: this method supposed to return `Dimension` object (even according the comment above).
And we already have `get_length()` support for `Dimension`:
https://github.com/openvinotoolkit/openvino_contrib/blob/b0826c595d60616a7253382e3b04a4023f73b803/modules/java_api/src/main/java/org/intel/openvino/Dimension.java#L20
https://github.com/openvinotoolkit/openvino_contrib/blob/b0826c595d60616a7253382e3b04a4023f73b803/modules/java_api/src/main/cpp/dimension.cpp#L12